src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -141,12 +144,21 @@ private static ExternalProjectReference GetExternalProject(PackageSpec rootProje
                 restoreArgs.Log);
             
             var rootPath = Path.GetDirectoryName(project.PackageSpec.FilePath);
+            var signedPackageVerifier = new PackageSignatureVerifier(SignatureVerificationProviderFactory.GetSignatureVerificationProviders());
```

Comment:
This feels really weird in the DependencyGraphSpecRequestProvider. 

Maybe it should be part of the RestoreArgs, and just set to the RestoreRequest here.

Reply:
I need to create this somewhere that has access to `ISettings`, it seems like here it gets a `settings` object in line 136. If I do it in `RestoreRequest` I would need to have access to `ISettings` as well.

Reply:
I can do something similar to [this](https://github.com/NuGet/NuGet.Client/pull/2428/files#diff-be8300f82028ee502433b49799fb34caR137), basically, have a method in `restoreArgs` that receives an `ISettings` and returns the `PackageExtractionContext`. Do you think that would be a good solution?